content/docs/lists-and-keys.md

```diff
@@ -119,34 +118,34 @@ const todoItems = todos.map((todo) =>
 );
 ```
  
-When you don't have stable IDs for rendered items, you may use the item index as a key as a last resort:
+Kiedy nie będziesz miał stabilnych identyfikatorów dla wyświetlanych elementów, w skrajnym przypadku możesz użyć indeksu jako klucz:
  
 ```js{2,3}
 const todoItems = todos.map((todo, index) =>
-  // Only do this if items have no stable IDs
+  // Rób to tylko w przypadku gdy elementy nie mają stabilnego identyfikatorów
   <li key={index}>
     {todo.text}
   </li>
 );
 ```
  
-We don't recommend using indexes for keys if the order of items may change. This can negatively impact performance and may cause issues with component state. Check out Robin Pokorny's article for an [in-depth explanation on the negative impacts of using an index as a key](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). If you choose not to assign an explicit key to list items then React will default to using indexes as keys.
+Nie zalecamy używania indeksów jako klucze jeżeli kolejność elementów może ulec zmianie. Może to negatywnie wpłynąć na wydajność i spowodować problemy ze stanem komponentu. Sprawdź artykuł Robin Pokorny na temat [szczegółowego wyjaśnienia negatywnych aspektów używania indeksu jako klucz](https://medium.com/@robinpokorny/index-as-a-key-is-an-anti-pattern-e0349aece318). Jeżeli nie zdecydujesz się jawnie ustawić kluczy dla elementów listy, React domyślnie użyje indeksów jako klucze.
  
-Here is an [in-depth explanation about why keys are necessary](/docs/reconciliation.html#recursing-on-children) if you're interested in learning more.
+Jeżeli chciałbyś dowiedzieć się wiecej, to tutaj znajdziesz [szczegółowe wyjaśnienie dlaczego klucze są niezbędne](/docs/reconciliation.html#recursing-on-children).
  
-### Extracting Components with Keys {#extracting-components-with-keys}
+### Wydrębnianie komponentów z kluczami {#extracting-components-with-keys}
  
-Keys only make sense in the context of the surrounding array.
+Klucze mają sens tylko w kontekście otaczającej tablicy.
  
-For example, if you [extract](/docs/components-and-props.html#extracting-components) a `ListItem` component, you should keep the key on the `<ListItem />` elements in the array rather than on the `<li>` element in the `ListItem` itself.
+Dla przykładu, jeżeli [wyodrębnisz](/docs/components-and-props.html#extracting-components) komponent `ListItem`, powinieneś trzymać klucz na elementach `<ListItem />`wewnątrz tablicy zamiast na elemencie `<li>` wewnątrz elementu `ListItem`.
  
-**Example: Incorrect Key Usage**
+**Przykłaď: Niepoprawne użycie klucza**
```

Comment:
```suggestion
**Przykład: niepoprawne użycie klucza**
```

Reply:
Według mnie to jest tytuł pewnej sekcji, więc powinniśmy zacząć go wielką literą.

Reply:
Też bym zostawił z dużej litery, chociaż nie jestem pewien czy tak jest poprawnie.
Może @marcinostrouch albo @cvqprs będzie wiedział? 🎉

Reply:
+1 dla wielkiej litery